torch_audiomentations/augmentations/spliceout.py

```diff
@@ -0,0 +1,124 @@
+import logging
+from turtle import forward
+import torch
+from typing import Optional
+from torch import Tensor
+from torch.nn.functional import pad
+
+from ..core.transforms_interface import BaseWaveformTransform
+from ..utils.dsp import convert_decibels_to_amplitude_ratio
+from ..utils.object_dict import ObjectDict
+
+
+class SpliceOut(BaseWaveformTransform):
+
+    """
+    spliceout augmentation proposed in https://arxiv.org/pdf/2110.00046.pdf
+    silence padding is added at the end to retain the audio length.
+    """
+
+    supported_modes = {"per_batch", "per_example"}
+    requires_sample_rate = True
+
+    def __init__(
+        self,
+        num_time_intervals=8,
+        max_width=25,
```

Comment:
Ideally, this default should be 400 ms, not 25 ms. The motivation here is that the authors of the paper used T=40 successfully, which in their case means 40 * 10 ms (because their spectrogram was made with a hop size of 10 ms)

https://arxiv.org/pdf/2110.00046.pdf

Reply:
Done.